RubberduckTests/Inspections/VariableNotUsedInspectionTests.cs

```diff
@@ -93,79 +68,77 @@ End Sub
     '@Ignore VariableNotUsed
     Dim var1 As String
 End Sub";
 
-            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
-            using (var state = MockParser.CreateAndParse(vbe.Object))
-            {
-
-                var inspection = new VariableNotUsedInspection(state);
-                var inspectionResults = inspection.GetInspectionResults(CancellationToken.None);
-
-                Assert.IsFalse(inspectionResults.Any());
-            }
+            Assert.AreEqual(0, GetTestResultCount(inputCode));
         }
 
         [Test]
         [Category("Inspections")]
-        public void VariableNotUsed_DoesNotReturnsResult_UsedInNameStatement()
+        public void VariableNotUsed_UsedInNameStatement_DoesNotReturnsResult()
         {
             const string inputCode =
-                @"Sub Foo()
+@"Sub Foo()
     Dim var1 As String
     Name ""foo"" As var1
 End Sub";
 
-            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
-            using (var state = MockParser.CreateAndParse(vbe.Object))
-            {
+            Assert.AreEqual(0, GetTestResultCount(inputCode));
+        }
 
-                var inspection = new VariableNotUsedInspection(state);
-                var inspectionResults = inspection.GetInspectionResults(CancellationToken.None);
+        //https://github.com/rubberduck-vba/Rubberduck/issues/5088        
+        [Test]
+        [Category("Inspections")]
+        public void VariableNotUsed_AssignedButNeverReferenced_ReturnsResult()
+        {
+            const string inputCode =
+@"Sub Foo()
+    Dim var1 As String
+    var1 = ""test""
+End Sub";
 
-                Assert.IsFalse(inspectionResults.Any());
-            }
+            Assert.AreEqual(1, GetTestResultCount(inputCode));
         }
 
         [Test]
         [Category("Inspections")]
-        public void VariableUsed_DoesNotReturnResultIfAssigned()
+        public void VariableNotUsed_AssignedMultpleTimesButNeverReferenced_ReturnsResult()
         {
             const string inputCode =
-                @"Function Foo() As Boolean
-    Dim var1 as String
-    var1 = ""test""
-End Function";
+@"Public Sub Foo()
+    Dim var2 As Long
+    var2 = 4
+    var2 = 7
+    var2 = 8
+End Sub";
 
+            Assert.AreEqual(1, GetTestResultCount(inputCode));
+        }
+
+        private int GetTestResultCount(string inputCode)
+        {
+            var resultCount = 0;
             var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
             using (var state = MockParser.CreateAndParse(vbe.Object))
             {
 
                 var inspection = new VariableNotUsedInspection(state);
                 var inspectionResults = inspection.GetInspectionResults(CancellationToken.None);
 
-                Assert.AreEqual(0, inspectionResults.Count());
+                resultCount = inspectionResults.Count();
             }
+            return resultCount;
         }
```

Comment:
Thanks for making this a bit more dry.
Could I ask you to further refactor this and base this test class on the recently introduced `InspectionTestBase`.

Reply:
Done